master/internal/rm/kubernetesrm/pods.go

```diff
@@ -1297,6 +1297,13 @@ func (p *pods) getNodeResourcePoolMapping(nodeSummaries map[string]model.AgentSu
 				}
 			}
 
+			// add default toleration so that autoscaling nodes will still be counted.
+			poolTolerations = append(poolTolerations, k8sV1.Toleration{
```

Comment:
@eecsliu now that i'm thinking, is there a way to make this code less fragile against taints we don't expect?